php/url/class-cloudinary.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Cloudinary class for the Cloudinary plugin.
+ *
+ * @package Cloudinary
+ */
+
+namespace Cloudinary\URL;
+
+use Cloudinary\Media;
+use function Cloudinary\get_plugin_instance;
+
+/**
+ * Class Cloudinary
+ */
+class Cloudinary extends Url_Object {
+
+	/**
+	 * Holds the Connect API object.
+	 *
+	 * @var \Cloudinary\Connect\Api
+	 */
+	protected $api;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string|null $url The optional raw URL.
+	 */
+	public function __construct( $url = null ) {
+		$this->api = get_plugin_instance()->get_component( 'connect' )->api;
+		parent::__construct( $url );
+	}
+
+	/**
+	 * Get cloudinary transformations.
+	 *
+	 * @return array
+	 */
+	public function get_transformations() {
+		static $transformations;
+		if ( ! $transformations && ! empty( $this->relation['transformations'] ) ) {
+			$transformations = Media::extract_transformations_from_string( $this->relation['transformations'], $this->get_type() );
+		}
+
+		return $transformations;
+	}
+
+	/**
+	 * Get a cloudinary transformation.
+	 *
+	 * @param string $transformation The transformation to get.
+	 *
+	 * @return array|string
+	 */
+	public function get_transformation( $transformation ) {
+		static $fetched = array();
+		if ( ! isset( $fetched[ $transformation ] ) ) {
+			$transformations = $this->get_transformations();
+			$matched         = null;
+			foreach ( $transformations as $transformation_set ) {
+				if ( isset( $transformation_set[ $transformation ] ) ) {
+					$matched = $transformation_set[ $transformation ];
+					break;
+				}
+			}
+			$fetched[ $transformation ] = $matched;
+		}
+
+		return $fetched[ $transformation ];
+	}
+
+	/**
+	 * Get the URL version.
+	 *
+	 * @return string
+	 */
+	public function get_version() {
+		static $version;
+		if ( ! $version ) {
+			$version = 'v1';
+			if ( preg_match( '/\/(v\d*)\//', $this->raw_url, $matches ) ) {
+				$version = $matches[1];
+			}
+		}
+
+		return $version;
+	}
+
+	/**
+	 * Get the public id of the url.
+	 *
+	 * @return string
+	 */
+	public function get_id() {
+		return $this->relation['public_id'] . '.' . $this->relation['format'];
+	}
+
+	/**
+	 * Get WordPress URL
+	 *
+	 * @return WordPress
+	 */
+	public function get_wordpress_url() {
+		static $object = array();
+		if ( empty( $object ) ) {
+			$url_component = get_plugin_instance()->get_component( 'url' );
+			$id            = $this->get_attachment_id();
```

Comment:
<!-- potential concern, ask for clarification or investigation -->
**❓question:** @DavidCramer The IDE is complaining that it couldn't find `get_attachment_id()` so didn't I. Can you please check this one closely?

Reply:
Alongside it, it also points out that both `get_transformation()` and `get_wordpress_url()` aren't in use.